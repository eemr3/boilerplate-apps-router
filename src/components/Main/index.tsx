const Main = () => (
  <main>
    <h1>React avancaÃ§o com a Main</h1>
  </main>
);

export default Main;
