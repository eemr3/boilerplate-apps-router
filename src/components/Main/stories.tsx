import { Meta, StoryObj } from '@storybook/react';
import Main from '.';

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    title: { control: 'text' },
    description: { control: 'text' },
  },
} as Meta;

export const Default: StoryObj = {};
