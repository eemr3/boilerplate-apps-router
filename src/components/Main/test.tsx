import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderizar o componente
    const { container } = render(<Main />);
    // buscar o elemento e verificar se ele existe
    expect(screen.getByRole('heading', { name: /react avanÃ§ado/i })).toBeInTheDocument();

    //gerar snapshot
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
});
