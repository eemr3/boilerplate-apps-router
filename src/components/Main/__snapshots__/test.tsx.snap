// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-braxZu cgRXPr"
>
  <h1>
    React avan√ßado
  </h1>
</main>
`;
